### ISSUE TITLE
Block for inclusion of Bluetooth device information service needed

### ISSUE THREAD

**Issue by bluetooth-mdw** [Microsoft/pxt-microbit#109]
It would be great if there was a tray containing Bluetooth related blocks.

In this tray, there should/could be a block per Bluetooth "service".

Dragging a Bluetooth service block into a PXT app would result in the corresponding Bluetooth being included in the generated code.

This issue requests a block for the micro:bit **Bluetooth device information service**.

Note that currently this service is included by default in builds. Given memory constraints it would make sense if it was only included when needed, either as a consequence of using a block which relies upon it (e.g. user events) or because it was explicitly included by dragging a new block into the code.

The Bluetooth device information service is included by setting a configuration flag in config.json. See device_info_service property below.

```
        "bluetooth":{
            "enabled": 1,
            "pairing_mode": 1,
            "private_addressing": 0,
            "open": 0,
            "whitelist": 1,
            "advertising_timeout": 0,
            "tx_power": 0,
            "dfu_service": 1,
            "event_service": 1,
            "device_info_service": 1
        },
```

Note that inclusion of Bluetooth services has consequences for memory configuration on the micro:bit, specifically with respect to GATT table size. I've suggested an approach to addressing this to Joe Finney of Lancaster University and we're evaluating. Will update issue once this discussion has progressed further.

**Comment by pelikhan** [comment]
Is there any C++ associated to this service?

**Comment by bluetooth-mdw** [comment]
No, the runtime instantiates it and DFU and event services based on flags in the config. They're deemed 'core services' for historical reasons only.

In the runtime code:

```C/C++
    // Bring up core BLE services.
#if CONFIG_ENABLED(MICROBIT_BLE_DFU_SERVICE)
    new MicroBitDFUService(*ble);
#endif

#if CONFIG_ENABLED(MICROBIT_BLE_DEVICE_INFORMATION_SERVICE)
    DeviceInformationService ble_device_information_service (*ble, MICROBIT_BLE_MANUFACTURER, MICROBIT_BLE_MODEL, serialNumber.toCharArray(), MICROBIT_BLE_HARDWARE_VERSION, MICROBIT_BLE_FIRMWARE_VERSION, MICROBIT_BLE_SOFTWARE_VERSION);
#endif

#if CONFIG_ENABLED(MICROBIT_BLE_EVENT_SERVICE)
    new MicroBitEventService(*ble, messageBus);
#else
    (void)messageBus;
#endif
```
